Tests/OpenAPIGeneratorReferenceTests/Resources/ReferenceSources/Petstore/Types.swift

```diff
@@ -154,25 +154,87 @@ extension APIProtocol {
 /// Server URLs defined in the OpenAPI document.
 public enum Servers {
     /// Example Petstore implementation service
+    public enum Server1 {
+        public static func url() throws -> Foundation.URL {
+            try Foundation.URL(
+                validatingOpenAPIServerURL: "https://example.com/api",
+                variables: []
+            )
+        }
+    }
+    /// Example Petstore implementation service
+    @available(*, deprecated, message: "Migrate to the new type-safe API for server URLs.")
     public static func server1() throws -> Foundation.URL {
         try Foundation.URL(
             validatingOpenAPIServerURL: "https://example.com/api",
             variables: []
         )
     }
+    public enum Server2 {
+        public static func url() throws -> Foundation.URL {
+            try Foundation.URL(
+                validatingOpenAPIServerURL: "/api",
+                variables: []
+            )
+        }
+    }
+    @available(*, deprecated, message: "Migrate to the new type-safe API for server URLs.")
     public static func server2() throws -> Foundation.URL {
         try Foundation.URL(
             validatingOpenAPIServerURL: "/api",
             variables: []
         )
     }
     /// A custom domain.
+    public enum Server3 {
+        /// The "port" variable defined in the OpenAPI document. The default value is "443".
+        @frozen public enum Port: Swift.String {
+            case _443 = "443"
+            case _8443 = "8443"
+        }
+        ///
+        /// - Parameters:
+        ///   - _protocol:
+        ///   - subdomain: A subdomain name.
+        ///   - port:
+        ///   - basePath: The base API path.
+        public static func url(
+            _protocol: Swift.String = "https",
+            subdomain: Swift.String = "test",
+            port: Port = Port._443,
```

Comment:
```suggestion
            port: Port = ._443,
```

Can we omit repeating the type, to match what users would type manually?

Reply:
I had 2 reasons for having it there
1. **Consistency.** Other types have been explicit (e.g. `Foundation.URL` and `Swift.String`) which I figured was more to disambiguate if any types conflicted, but doing it here kept it consistent with being explicit.
1. **Speed/Efficiency.** The compiler wouldn't need to use type inference and consume as many resources on the type checking; which given the other issues raised around compile speed I figured that being explicit would help on that front, especially on larger documents with a larger number of generated symbols.


Would you agree with my justification here?

Reply:
It's fine either way.

Re 1, we really do generate fully qualified type names in most places to avoid possibly clashing with generated types. But default values don't need to be fully qualified, see below.

Re 2, it's fine performance wise, as the type of the parameter is explicitly specified in the declaration, so the compiler doesn't have to infer anything, it just needs to verify the default value can satisfy the type.

If you see that we're generating a fully qualified name in a default parameter value anywhere, let me know, we might not need to do that.

Reply:
Interesting, I had seen compile performance issues previously with type inference on parameters; though it was quite a while ago now, so maybe it's been long fixed and I'm just stuck in the past.

I'll remove the explicit type from the parameter with my next update.